src/AutorestSwift/Code Generator/Logger.swift

```diff
@@ -87,8 +90,9 @@ class FileLogger {
         try? url.append(line: msg)
     }
 
-    func logAndFail(_ message: @autoclosure @escaping () -> String?) {
-        log(message())
-        fatalError(message() ?? "")
+    func logAndFail(_ message: @autoclosure @escaping () -> String?) -> Never {
```

Comment:
The `Never` return type avoids you having to `return` in guard statements when this is used.
```Swift
// THIS
guard let foo = bar else {
  logAndFail(...)
}

// Instead of THIS
guard let foo = bar else {
  logAndFail(...)
  return
}
```